src/loadsave.cpp

```diff
@@ -408,6 +433,32 @@ bool LoadGameFile(const char *fname)
 	}
 }
 
+/**
+ * Load basic data from a savegame file.
+ * @param fname Name of the file to load.
+ * @return Basic information about the savegame.
+ * @note Check the return value's #load_success attribute to see whether preloading was successful.
+ */
+PreloadData PreloadGameFile(const char *fname)
+{
+	PreloadData result;
+	if (fname == nullptr) return result;
+
+	try {
+		FILE *fp = fopen(fname, "rb");
+		if (fp == nullptr) return result;
+		Loader ldr(fp);
+		result = Preload(ldr);
+		fclose(fp);
+	} catch (...) {
```

Comment:
I was thought to avoid catching all exceptions, and instead only catch those that you really want to catch.
The reason for that is here you may grab totally unrelated exceptions that arise due to programming errors, and silently discard them. That means bugs may be hiding in here without ever being found.

The only exception to that rule is if you rethrow.